apis/cloudflare/src/tracing.ts

```diff
@@ -0,0 +1,187 @@
+): unknown {
+  if (isArray(data)) {
+    return data.map((item) => replacePayloadWithAttachmentsInner(item, state));
+  } else if (isObject(data)) {
+    return Object.fromEntries(
+      Object.entries(data).map(([key, value]) => [
+        key,
+        replacePayloadWithAttachmentsInner(value, state),
+      ]),
+    );
+  } else if (typeof data === "string") {
+    if (isBase64Image(data)) {
+      const { mimeType, data: arrayBuffer } = getBase64Parts(data);
+      const filename = `file.${mimeType.split("/")[1]}`;
+      return new Attachment({
+        data: arrayBuffer,
+        contentType: mimeType,
+        filename,
+        state,
+      });
+    } else {
+      return data;
+    }
+  } else {
+    return data;
+  }
+}
+
+const base64ImagePattern =
+  /^data:image\/[a-zA-Z]+;base64,[A-Za-z0-9+/]+={0,2}$/;
+export function isBase64Image(s: string): boolean {
+  // Avoid unnecessary (slower) pattern matching
+  if (!s.startsWith("data:")) {
+    return false;
+  }
+
+  return base64ImagePattern.test(s);
+}
+// Being as specific as possible about allowable characters and avoiding greedy matching
+// helps avoid catastrophic backtracking: https://github.com/braintrustdata/braintrust/pull/4831
+const base64ContentTypePattern =
+  /^data:([a-zA-Z0-9]+\/[a-zA-Z0-9+.-]+);base64,/;
+export function getBase64Parts(s: string): {
+  mimeType: string;
+  data: ArrayBuffer;
+} {
+  const parts = s.match(base64ContentTypePattern);
+  if (!parts) {
+    throw new Error("Invalid base64 image");
+  }
+  const mimeType = parts[1];
+  const data = s.slice(`data:${mimeType};base64,`.length);
+  return { mimeType, data: base64ToArrayBuffer(data) };
+}
+
+// XXX delete this once the new core version is live
+const EXPERIMENT_ID_PREFIX = "experiment_id:";
+const PROJECT_ID_PREFIX = "project_id:";
+const PROJECT_NAME_PREFIX = "project_name:";
+const PLAYGROUND_ID_PREFIX = "playground_id:";
+
+export function resolveParentHeader(header: string): SpanComponentsV3 {
+  if (header.startsWith(EXPERIMENT_ID_PREFIX)) {
+    return new SpanComponentsV3({
+      object_type: SpanObjectTypeV3.EXPERIMENT,
+      object_id: header.substring(EXPERIMENT_ID_PREFIX.length),
+    });
+  } else if (header.startsWith(PROJECT_ID_PREFIX)) {
+    return new SpanComponentsV3({
+      object_type: SpanObjectTypeV3.PROJECT_LOGS,
+      object_id: header.substring(PROJECT_ID_PREFIX.length),
+    });
+  } else if (header.startsWith(PLAYGROUND_ID_PREFIX)) {
+    return new SpanComponentsV3({
+      object_type: SpanObjectTypeV3.PLAYGROUND_LOGS,
+      object_id: header.substring(PLAYGROUND_ID_PREFIX.length),
+    });
+  } else if (header.startsWith(PROJECT_NAME_PREFIX)) {
+    const projectName = header.substring(PROJECT_NAME_PREFIX.length);
+    return new SpanComponentsV3({
+      object_type: SpanObjectTypeV3.PROJECT_LOGS,
+      compute_object_metadata_args: {
+        project_name: projectName,
+      },
+    });
+  }
+
+  return SpanComponentsV3.fromStr(header);
```

Comment:
should we catch the invalid encoding error here and ignore the header if its malformed?

Reply:
hmm interesting idea, is that the behavior you'd want?

Reply:
the error response you added also works, as long as we don't have some uncaught exception here and result in a 500 response.

Reply:
It ends up as a nice 400!